File: iOS/ObjectiveC/SimpleIntegration/SimpleIntegrationObjC/InterstitialAd/InterstitialAdViewController.m
//
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
//

#import "InterstitialAdViewController.h"
#import <AppNexusSDK/AppNexusSDK.h>

@interface InterstitialAdViewController () <ANInterstitialAdDelegate>{
    UIActivityIndicatorView *indicator;
}

@property (strong, nonatomic) ANInterstitialAd *interstitialAd;

@end

@implementation InterstitialAdViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = @"Interstitial Ad";
    self.interstitialAd = [[ANInterstitialAd alloc] initWithPlacementId:@"17058950"];
    self.interstitialAd.delegate = self;
    self.interstitialAd.clickThroughAction = ANClickThroughActionReturnURL;
    [self.interstitialAd loadAd];
}

#pragma mark - ANInterstitialAdDelegate

- (void)adDidReceiveAd:(id)ad {
    NSLog(@"adDidReceiveAd");
    [self.interstitialAd displayAdFromViewController:self];
}

-(void)ad:(id)ad requestFailedWithError:(NSError *)error{
    NSLog(@"Ad request Failed With Error");
}
@end

File: iOS/Swift/SimpleIntegration/SimpleIntegrationSwift/InterstitialAd/InterstitialAdViewController.swift
//
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
//

import UIKit
import AppNexusSDK

class InterstitialAdViewController: UIViewController , ANInterstitialAdDelegate {
    
    var interstitialAd: ANInterstitialAd?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "Interstitial Ad"
        
        interstitialAd = ANInterstitialAd(placementId: "17058950")
        interstitialAd!.delegate = self
        interstitialAd!.clickThroughAction = ANClickThroughAction.openSDKBrowser
        interstitialAd!.load()
        // Do any additional setup after loading the view.
    }
    
    // MARK: - ANInterstitialAdDelegate
    func adDidReceiveAd(_ ad: Any) {
        print("adDidReceiveAd")
        interstitialAd!.display(from: self)
    }
    
    func ad(_ ad: Any, requestFailedWithError error: Error) {
        print("Ad request Failed With Error")
    }
}

